squashfs-tools/xz_wrapper.c

```diff
@@ -140,6 +148,80 @@ static int xz_options(char *argv[], int argc)
 			}
 		}
 
+		return 1;
+	} else if(strcmp(argv[0], "-Xpreset") == 0) {
+		char *b;
+		long val;
+
+		if(argc < 2) {
+			fprintf(stderr, "xz: -Xpreset missing preset-level (valid value 0-9)\n");
+			goto failed;
+		}
+
+		val = strtol(argv[1], &b, 10);
+		if ((int) val < 0 || (int) val & ~LZMA_PRESET_LEVEL_MASK) {
+			fprintf(stderr, "xz: -Xpreset can't be "
+				"negative or more than the max preset\n");
+			goto failed;
+		}
+
+		preset = (int) val;
```

Comment:
What happens if we specify `-Xe` first and then `-Xpreset 6`?

Reply:
Good catch! I will change this to reset the values and OR the new preset.